system/HTTP/RedirectResponse.php

```diff
@@ -93,8 +93,8 @@ public function withInput()
     {
         $session = service('session');
         $session->setFlashdata('_ci_old_input', [
-            'get'  => $_GET ?? [],
-            'post' => $_POST ?? [],
+            'get'  => is_array($_GET) ? $_GET : [],
+            'post' => is_array($_POST) ? $_POST : [],
```

Comment:
`$_GET` or `$_POST` already return empty array.

```suggestion
            'get'  => $_GET,
            'post' => $_POST,
```

Ref : https://3v4l.org/u4dsh